src/chainparamsbase.cpp

```diff
@@ -63,41 +60,31 @@ class CBaseRegTestParams : public CBaseChainParams
         strDataDir = "regtest";
     }
 };
-static CBaseRegTestParams regTestParams;
-
-/*
- * Unit test
- */
-class CBaseUnitTestParams : public CBaseMainParams
-{
-public:
-    CBaseUnitTestParams()
-    {
-        strDataDir = "unittest";
-    }
-};
-static CBaseUnitTestParams unitTestParams;
 
-static CBaseChainParams* pCurrentBaseParams = 0;
+static Container<CBaseChainParams> cGlobalChainBaseParams;
 
 const CBaseChainParams& BaseParams()
 {
-    assert(pCurrentBaseParams);
-    return *pCurrentBaseParams;
+    return cGlobalChainBaseParams.Get();
 }
 
-void SelectBaseParams(const std::string& chain)
+CBaseChainParams* CBaseChainParams::Factory(const std::string& chain)
```

Comment:
Using strings instead of enums for chain ~~identification~~ selection (for identification we should use the genesis block) it's not something that's being changed here, it was changed in #6235. Apart from the simplifications in that PR (and the fact that this very PR would be more complicated without having done #6235 first), here's a few other commits in #6382 that get simpler/more consistent by using strings instead of an enum: https://github.com/jtimon/bitcoin/commit/453597a3897a22ef45aa132ac285ccd71eba2397 https://github.com/jtimon/bitcoin/commit/ff440ce6f7e20a410d43cbf111d49e53ab497eb8 https://github.com/jtimon/bitcoin/commit/d644c2993454d29ee0d9c26bc9e107831e1e5b44 https://github.com/jtimon/bitcoin/commit/b7bce5c5571200361e3b74406d0f9bc986960491